client/src/pages/SingleProduct.js

```diff
@@ -0,0 +1,277 @@
+import React, { useEffect, useState, useContext } from "react";
+import { makeStyles } from "@material-ui/core/styles";
+import { BrowserRouter as Router, useParams } from "react-router-dom";
+import Grid from "@material-ui/core/Grid";
+import Box from "@material-ui/core/Box";
+import Typography from "@material-ui/core/Typography";
+import Skeleton from "@material-ui/lab/Skeleton";
+import { fetchGet } from "../utils/ApiFetching";
```

Comment:
I feel like this could have been better designed - but this is something that we can fix later 

As opposed to having a fetchGet,  and a fetchPost, if would have been nicer to just generalize it and have an interface like:
api('get', ..) or api.get(...) like the fetch library itself

Reply:
I will work on this later.